helper/hex/hex.go

```diff
@@ -87,9 +87,14 @@ func EncodeBig(bigint *big.Int) string {
 }
 
 // DecodeHexToBig converts a hex number to a big.Int value
-func DecodeHexToBig(hexNum string) *big.Int {
+func DecodeHexToBig(hexNum string) (*big.Int, error) {
 	createdNum := new(big.Int)
-	createdNum.SetString(hexNum, 16)
+	cleaned := strings.TrimPrefix(hexNum, "0x")
 
-	return createdNum
+	_, ok := createdNum.SetString(cleaned, 16)
+	if !ok {
+		return nil, fmt.Errorf("failed to convert string: %s to big.Int with base: 16", hexNum)
+	}
+
+	return createdNum, nil
```

Comment:
Spare 1 line :smile:

```suggestion
	cleaned := strings.TrimPrefix(hexNum, "0x")

	value, ok := new(big.Int).SetString(cleaned, 16)
	if !ok {
		return nil, fmt.Errorf("failed to convert string: %s to big.Int with base: 16", hexNum)
	}

	return value, nil
```

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/1289/commits/8513e9d2713d95a0bbb1df4eed52c8d0422cc6a0